### PR TITLE
Single Log Out support

### PR DESCRIPTION
**Pull request by ralequi**
Added requested feature on #58 .

This commit changes and adds many things that have been required to implement the single sign out:

- Added some defaulted settings to simplify configuration (see [simple_sso/settings.py](simple_sso/settings.py)):
- Added APP `simple_sso.sso_client` to send client logouts to server.
- Added Middleware `simple_sso.sso_client.middleware.PostAuthenticationMiddleware` to have a "keep-alive" behaviour and checks every X seconds if the current user is still connected on the server. (Defaulted to 60 but can be 0 to check on every client request)
- Fixed tests for Django 3.x

I think the usage is clear on the modified README. Please, tell me if it's not clear enough or you see any code-issue.

Everything was tested locally and with a couple of remote servers.
Testing environment:
- python 3.8 (ubuntu20) & python 3.9 (Fedora)
- Django 3.2.3 (ubuntu20) & 3.2.5 (Fedora)

### CODE REVIEW

File: simple_sso/sso_client/models.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+
+from simple_sso.settings import settings
+from django.urls import NoReverseMatch, reverse
+
+from webservices.sync import SyncConsumer
+
+
+def logout_sso_client(sender, request, user, **kwargs):
```
Comment (paradxum) [inline simple_sso/sso_client/models.py:8]: Why would this receiver be in a "models" file? It does not seem to be a model/data definition. Perhaps utilize a "signals.py" file

File: simple_sso/sso_server/models.py [hunk 2]
```diff
@@ -52,6 +53,16 @@ def rotate_keys(self):
         self.save()
 
 
+def logout_token(sender, request, **kwargs):
```
Comment (paradxum) [inline simple_sso/sso_server/models.py:56]: Same as with client... Why would this receiver be in a "models" file? It does not seem to be a model/data definition. Perhaps utilize a "signals.py" file

File: simple_sso/sso_server/server.py [hunk 3]
```diff
@@ -1,19 +1,29 @@
...
 from django.views.generic.base import View
 from itsdangerous import URLSafeTimedSerializer
+from simple_sso.settings import settings
 from simple_sso.sso_server.models import Token, Consumer
-import datetime
 from webservices.models import Provider
 from webservices.sync import provider_for_django
 
 
+class ThrowableHttpResponse(Exception):
```
Comment (paradxum) [inline simple_sso/sso_server/server.py:19]: This seems to be here to capture and ignore errors (great for debugging but not production.) Please let me know if I'm missing something!

**Comment by AlessioBazzanella** [comment]
Any update here? Thanks

**Comment by derlin** [comment]
Hello @ralequi and thank you for this awesome contribution!

However, as we already mentioned in #64 , this package is being used internally in code paths in our products that we are looking into deprecating, and we have decided to not maintain this repository anymore. We will thus not have the bandwidth necessary to ensure a thorough security process, as required for such a delicate package.

@AlessioBazzanella It seems ralequi forked this package at https://github.com/naudit/django-simple-sso (and pipy https://pypi.org/project/django-simple-sso.naudit/), so I would suggest moving this issue to the fork, where it has a higher chance of being integrated.

Thank you again for your interest and contribution, and happy coding!

**Pull request closed without merging**